apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx

```diff
@@ -38,63 +38,51 @@ import { tagToBadgeData } from './consts';
 export const Actions: FunctionComponent<IActionsProps> = ({
   id,
   fullName,
+  avatarUrl,
   showResolutionButtons,
 }) => {
   const {
-    onMutateApproveEntity,
-    onMutateRevisionCase,
-    onMutateRejectEntity,
-    onMutateAssignWorkflow,
-    debouncedIsLoadingApproveEntity,
-    debouncedIsLoadingRejectEntity,
-    debouncedIsLoadingRevisionCase,
+    tag,
+    assignedUser,
+    authenticatedUser,
     isLoading,
     isLoadingCase,
     canApprove,
     canReject,
     canRevision,
-    caseState,
-    tag,
-    authenticatedUser,
     assignees,
-    onTriggerAssignToMe,
-    hasDecision,
+    debouncedIsLoadingApproveEntity,
+    debouncedIsLoadingRejectEntity,
+    debouncedIsLoadingRevisionCase,
     documentsToReviseCount,
+    onMutateApproveEntity,
+    onMutateRevisionCase,
+    onMutateRejectEntity,
+    onMutateAssignWorkflow,
   } = useCaseActionsLogic({ workflowId: id, fullName });
 
   return (
     <div className={`sticky top-0 z-50 col-span-2 space-y-2 bg-base-100 px-4 pt-4`}>
-      <div className={`flex flex-row space-x-3.5`}>
-        <AssignButton
+      <div className={`mb-8 flex flex-row space-x-3.5`}>
+        <AssignDropdown
+          assignedUser={assignedUser}
+          avatarUrl={avatarUrl}
           assignees={[
             {
               id: authenticatedUser?.id,
               fullName: authenticatedUser?.fullName,
             },
+            ...((assignees ?? []) as Assignee[]),
           ]}
-          authenticatedUser={authenticatedUser}
-          caseState={caseState}
           onAssigneeSelect={id => {
-            onMutateAssignWorkflow(id, onTriggerAssignToMe);
+            onMutateAssignWorkflow(id, id === authenticatedUser?.id);
```

Comment:
Let's revisit this flag.

Reply:
We should fix it in a separate (*cough cough* tech debt *cough cough*) issue